sdk/src/message.rs

```diff
@@ -198,7 +196,7 @@ impl Message {
         } = get_keys(&instructions, payer);
         let num_required_signatures = signed_keys.len() as u8;
         signed_keys.extend(&unsigned_keys);
-        let instructions = compile_instructions(instructions, &signed_keys);
+        let instructions = compile_instructions(&instructions, &signed_keys);
```

Comment:
Added `&` in this line is unneeded (also can remove `&` from the call to `get_keys()` immediately above)

Reply:
Thanks, I must have made those changes before I ended up changing `new_with_payer`, suprised that clippy didn't complain.

Reply:
Yeah, it is surprisingly permissive of double references in some contexts, and we have a quite a few in our code base. Discovered that in one of the more heavy-handed Signer trait-object iterations.